File: 02-programacion-basica/14-depuracion-y-pruebas/java/practicas/PracticaValidarPromedio.java
public class PracticaValidarPromedio {
    public static void main(String[] args) {
        comprobar(new int[]{10, 20}, 15.0);
        comprobar(new int[]{20}, 20.0);
        comprobar(new int[]{}, 0.0);
    }

    static double promedio(int[] valores) {
        if (valores.length == 0) return 0;
        int suma = 0;
        for (int valor : valores) suma += valor;
        return (double) suma / valores.length;
    }

    static void comprobar(int[] entrada, double esperado) {
        double obtenido = promedio(entrada);
        // Double.compare evita depender de una comparación directa poco clara entre decimales.
        boolean correcto = Double.compare(esperado, obtenido) == 0;
        System.out.printf("esperado=%.2f | obtenido=%.2f | %s%n",
                esperado, obtenido, correcto ? "OK" : "ERROR");
    }
}

File: 02-programacion-basica/14-depuracion-y-pruebas/scala/practicas/PracticaValidarPromedio.scala
def promedioPrueba(valores: List[Int]): Double =
  if valores.isEmpty then 0.0 else valores.sum.toDouble / valores.size

@main def practicaValidarPromedio(): Unit =
  val casos = List(
    (List(10, 20), 15.0),
    (List(20), 20.0),
    (List.empty[Int], 0.0)
  )

  for (entrada, esperado) <- casos do
    val obtenido = promedioPrueba(entrada)
    // La prueba compara el resultado real con un valor conocido de antemano.
    println(s"esperado=$esperado | obtenido=$obtenido | ${if esperado == obtenido then "OK" else "ERROR"}")
